lib/httpClient.ts

```diff
@@ -43,4 +55,53 @@ export default abstract class HTTPClient {
             },
         );
     }
+
+    // https://github.com/axios/axios/issues/1666
+    checkLimit(config: AxiosRequestConfig) {
+        let limitType = LimitType.rate;
+        if (config.url?.endsWith('/ingest')) {
+            limitType = LimitType.ingest;
+        } else if (config.url?.endsWith('/query') || config.url?.endsWith('/_apl')) {
+            limitType = LimitType.query;
+        }
+
+        let limit: Limit = {
+            scope: LimitScope.unknown,
+            type: limitType,
+            value: 0,
+            remaining: -1,
+            reset: 0,
+        };
+        let foundLimit = false;
+        for (let scope of Object.values(LimitScope)) {
+            const key = `${scope}:${limitType}`;
+            if (this.limits[key]) {
+                limit = this.limits[key];
+                foundLimit = true;
+                break;
+            }
+            
+        }
+
+        // create fake response
+        const now = new Date();
+        const timestampInSeconds = Math.floor(now.getTime() / 1000);
+        if (foundLimit && limit.remaining == 0 && timestampInSeconds < limit.reset) {
+            config.adapter = (config) =>
+                new Promise((_, reject) => {
+                    const res: AxiosResponse = {
+                        data: `${limit.scope} ${limitType.toString()} limit exceeded, not making remote request`,
```

Comment:
How does the error message look like if `limit.scope` is not populated (e.g. the request wasn't an `api` but `ingest`/`query` request)?

Reply:
the limit scope always start being `unknown`.